bin/release.php

```diff
@@ -0,0 +1,283 @@
+<?php
+/**
+ * Plugin release management.
+ *
+ * Example usage: `php bin/release.php`
+ *
+ * @package jetpack
+ */
+
+/**
+ * Functions!
+ */
+
+function build_or_update_production_release_branch( $version ) {
+	$tmp = '/tmp/build-release';
+	execute_command( sprintf( 'rm -rf %s', escapeshellarg( $tmp ) ), 'Could not clean.' );
+
+	$release_branch      = "release/$version";
+	$release_branch_prod = "release/$version-prod";
+
+	execute_command( sprintf( 'git checkout %s && git pull', escapeshellarg( $release_branch ) ), 'Could not check out to release branch.' );
+
+	needs_action( "Building production branch...\n" );
+	execute_command( 'yarn build-production', 'Something went wrong. See output above for error.' );
+	success( "Release built! Now purging dev files for built branch...\n" );
+	purge_dev_files();
+
+	// Create a new local branch if none exists, else checkout to it
+	$branch_exists = execute_command( sprintf( 'git ls-remote --exit-code --heads origin %s', escapeshellarg( $release_branch_prod ) ), '', true );
+	if ( empty( $branch_exists ) ) {
+		execute_command( sprintf( 'git checkout -b %s', escapeshellarg( $release_branch_prod ) ), "Could not create local release branch: $release_branch_prod" );
+	} else {
+		$remote_url = execute_command( 'git remote get-url --all origin', 'Error', true );
+		execute_command( sprintf( 'git clone --depth 1 -b %1s --single-branch %2s %3s', escapeshellarg( trim( $release_branch_prod ) ), escapeshellarg( $remote_url ), escapeshellarg( $tmp ) ), 'Could not do it!' );
+		execute_command( sprintf( 'rsync -r --delete --exclude="*.git*" . %s', $tmp ), 'Could not rsync.' );
+		chdir( $tmp );
+	}
+
+	// Commit and push!
+	execute_command( 'git commit -am "New Build"', 'Could not commit.' );
+	execute_command( sprintf( 'git push -u origin %s', escapeshellarg( $release_branch_prod ) ), "Could not push $release_branch_prod to remote." );
+}
+
+/**
+ * Resets everything.
+ */
+function reset_and_clean() {
+	execute_command( 'git fetch origin && git checkout master && git reset --hard origin/master' );
+	execute_command( 'rm -rf /tmp/build-release' );
+}
+
+function purge_dev_files() {
+	$ignored = file( dirname( dirname( __FILE__ ) ) . '/.svnignore', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES );
+
+	foreach ( $ignored as $file_pattern ) {
+		execute_command( sprintf( 'rm -rf %s', $file_pattern ) );
```

Comment:
How about allowing us to add comments to the file, that won't be processed, so we can explain what the file is used for?

```suggestion
		if ( ! preg_match( '/^#/', $file_pattern ) ) {
			execute_command( sprintf( 'rm -rf %s', $file_pattern ) );
		}
```

Reply:
good call!